packages/react/src/components/form/initial.tsx

```diff
@@ -157,18 +167,42 @@ const HandleForm: React.FC<HandleFormProps> = ({
         label={text["initial.handle.email"]}
         placeholder={text["initial.handle.phone.email"]}
         value={values[handleType] ?? ""}
-        onChange={registerField(handleType, (value) => {
-          if (!isValidEmail(value)) {
-            return { message: text["validationError.email"] };
-          }
+        onChange={registerField(handleType, {
+          defaultValue:
+            shouldStoreLastID &&
+            lastIdentifier.handleType === "email_address" &&
+            lastIdentifier.value
+              ? lastIdentifier.value
+              : undefined,
+          validator: (value) => {
+            if (!isValidEmail(value)) {
+              return { message: text["validationError.email"] };
+            }
+          },
         })}
       />
     );
-  }, [flag, handleType, text, registerField, values]);
+  }, [
+    flag,
+    handleType,
+    text,
+    registerField,
+    values,
+    lastIdentifier.handleType,
+    lastIdentifier.value,
+    shouldStoreLastID,
+  ]);
 
   const onSubmit: React.FormEventHandler<HTMLFormElement> = (e) => {
     e.preventDefault();
 
+    if (shouldStoreLastID) {
```

Comment:
Should this be driven from the UI? Most of the business logic should be implemented either in the `flowState` or the SlashId context.